wordpress-mobile/WordPress-Android
# Unified Dashboard: Quick Start Support

Parent https://github.com/wordpress-mobile/WordPress-Android/issues/19270 

This PR refactors quick start to support the more menu

**Main Changes**
- Add QuickStartMenuStep
- Update `QuickStartRepository` to include a new mutableLiveData<QuickStartMenuStep?> which will be observed in `MySiteViewModel` and passed along to `QuickLinksItemViewModelSlice` and used to show the More focus point
- Update to show the "tap more menu" snackbar message
- Show the QS focus pt in the More menu on the correct item
- Update `QuickStartRepository.setActiveTask` to take in a boolean indicating if this is being called from the more menu. This prevents it from looping again through task being set to pending instead of active
- Implement a stop-gap solution to support showing a snackbar message in the More Menu. Added a "compose" callback within `SnackbarSequencer` that will post the snackbar back to listeners. This will be picked up in `MenuActivity`; which will create a lightweight message to be displayed. 
- Include a delay in MediaGridFragment and PublicizeListFragment show snackbar methods to account for the fact that the EventBus message gets there very quickly.


**NOTES**: 
- By the time the callback is invoked within `SnackbarSequencer`, the icon has already been merged with the message and there is no opportunity to pull it out. 
- The quick start next task won't prompt the user when in the more view. It is expected that the user tap the quick start card to continue if they had to leave for the More Menu. Perhaps we can move the order of the tasks, so the More menu taps are last?
- A non-stop gap solution would include completely rewriting the sequencer and the data classes, so that compose and traditional snackbars can be handled properly. Perhaps a `SnackbarManager` interface an two implementations:`ComposeSnackbarManager` and `TraditionalSnackbarManager`. The managers would implement their own logic for creating snackbars, but use the single queue in `SnackbarSequencer`. The type of data class in the queue should change to something more generic so both managers can create the proper snackbar. 
- 

**To test:**
- Install the app (fresh install)
- Login and select a site
- When prompted with the quick start dialog, select "Show me around"
- ✅ Go through the Quick Start Card and make sure the focus points are correct


## Regression Notes
1. Potential unintended areas of impact
Quick Start does not work properly

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and added new unit tests

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

UI Changes testing checklist: N/A
